FPEngine.cc

```diff
@@ -149,6 +149,17 @@ extern "C" int g_has_npcap_loopback;
  * network resources and schedule packet transmissions. */
 FPNetworkControl global_netctl;
 
+/* To send decoy probe one global data structure of type map has been used.
+ * It stores packet which has unique packet id which is concatenation of 
+ * ProbeID + decoynumber + target_host's number*/
+static std::map<std::string, IPv6Header*> decoy_probes;
+
+std::string get_unique_id(const char *id, int decoynumber, const char *targetIP) {
+  std::string id_str(id), ip_str(targetIP);
+  std::stringstream out;
+  out<<id_str<<decoynumber<<ip_str;
```

Comment:
This isn't necessarily unique: since targetIP could start with a decimal digit, it's not clear where the delimiter between the `decoynumber` and the `targetIP` are. Also, several of the probe IDs end with decimal digits, further causing confusion. Either choose a non-printable delimiter (like `"\x1C"`, which is used by Perl and Awk for this purpose) or rethink the use of a map structure here. (more thoughts on that later)